tracer/src/Datadog.Trace/TraceProcessors/NormalizerTraceProcessor.cs

```diff
@@ -0,0 +1,189 @@
+                +----------+--------------------------------------------------------------------------------------------------------+
+                | Service  | If empty, it gets set to a default.                                                                    |
+                | Service  | If too long, truncated at 100 characters.                                                              |
+                | Name     | If empty, set it to “unnamed_operation”.                                                               |
+                | Name     | If too long, truncated to 100 characters.                                                              |
+                | Resource | If empty, set to the same value as Name.                                                               |
+                | Duration | If smaller than 0, it is set to 0.                                                                     |
+                | Duration | If larger than math.MaxInt64 - Start, it is set to 0.                                                  |
+                | Start    | If smaller than Y2K, set to now - Duration or 0 if the result is negative.                             |
+                | Type     | If too long, truncated to 100.                                                                         |
+                | Meta     | “http.status_code” key is deleted if it’s an invalid numeric value smaller than 100 or bigger than 600 |
+                +----------+--------------------------------------------------------------------------------------------------------+
+             */
+
+            for (var i = trace.Offset; i < trace.Count + trace.Offset; i++)
+            {
+                trace.Array[i] = Process(trace.Array[i]);
+            }
+
+            return trace;
+        }
+
+        public Span Process(Span span)
+        {
+            // https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/agent/normalizer.go#L44-L56
+            span.ServiceName = NormalizeService(span.ServiceName);
+
+            // https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/agent/normalizer.go#L69-L80
+            span.OperationName = NormalizeName(span.OperationName);
+
+            // https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/agent/normalizer.go#L82-L86
+            if (string.IsNullOrEmpty(span.ResourceName))
+            {
+                Log.Information("Fixing malformed trace. Resource is empty (reason:resource_empty), setting span.resource={name}: {span}", span.OperationName, span);
+                span.ResourceName = span.OperationName;
+            }
+
+            // https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/agent/normalizer.go#L101-L105
```

Comment:
Can we leave some code comment here to indicate that we're skipping the ParentID check? I'm hoping this file can be read side-by-side with the trace agent normalizer so a comment here could make clear that we've decided not to implement it, not that we forgot about it